File: cpp/Two_Sum.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> memo;
        for(int i=0; i<nums.size(); i++) {
            if(memo.find(nums[i]) != memo.end()) return {memo[nums[i]], i};
            memo[target - nums[i]] = i;
        }
        return {};
    }
};

int main(){
    vector<int> inp = {2,7,11,15};
    Solution sol;
    vector<int> out = sol.twoSum(inp, 9);
    return 0;
}

File: ruby/two_sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  memo = {}
  nums.each_with_index do |num, idx|
    return [memo[num], idx] if memo[num]
    memo[target-num] = idx
  end
end

puts two_sum([3,2,4], 6)

